packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -195,6 +206,28 @@ function patchElement(n1: VElement, n2: VElement) {
     patchChildren(n1.children, n2.children, elm);
 }
 
+function mountStatic(vnode: VStatic, parent: ParentNode, anchor: Node | null) {
+    const { owner } = vnode;
+    const elm = (vnode.elm = cloneNode(vnode.fragment, true));
+
+    linkNodeToShadow(elm, owner);
+
+    // Marks this node as Static to propagate the shadow resolver. must happen after elm is assigned to the proper shadow
+    const { renderMode, shadowMode } = owner;
+
+    if (isSyntheticShadowDefined) {
+        if (shadowMode === ShadowMode.Synthetic || renderMode === RenderMode.Light) {
```

Comment:
```suggestion
        if (shadowMode === ShadowMode.Synthetic) {
```

Setting the `KEY__SHADOW_STATIC` should only be necessary for synthetic shadow, not light DOM. Light DOM doesn't need the shadow attributes. (If anything, it may need classes, which come from `*.scoped.css`. But those never change regardless of native, synthetic, light, or shadow – the class always needs to be added.)

Reply:
this is to set the shadow resolver (nodeOwner), which is used by synthetic shadow to know to which shadow the node belongs, so it needs to be set when is running on light mode.

Reply:
@jodarove I tried my change, and the tests all passed (jest, karma synthetic, karma native). Are we missing a test for this scenario?

Reply:
my thought process is, that if we can remove this, then we also can remove it [here](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/engine-core/src/framework/rendering.ts#L344-L348), once you remove it in `linkNodeToShadow`, there will be 2 failing tests. I can add a test for the case of static nodes.

Reply:
I see, so there is just a missing test?

Reply:
uff, i sweat to create [this test](https://github.com/salesforce/lwc/pull/2781/commits/38c28d7d554902388c8318b9bd20e1f6fd9427e1) jaja. The thing is that for most of our API, we check for the `nearestOwnerKey`, except for some (like `contains`). 

@caridy we could evaluate to modify those APIs to use `getNearestOwnerKey` and avoid setting `KEY__SHADOW_STATIC` (and do the traversal down), however, we need to evaluate the perf implications, since with this PR, we only do it once, vs (removing the down traversal) we will be doing it multiple times, (when accessing node owner on a static node). For now, i prefer to set the owner key in nodes.

Reply:
Well, that was my first choice @jodarove, to fix synthetic shadow to always traverse up when needing a nearest owner and remove the unnecessary loop.